lib/spack/spack/fetch_strategy.py

```diff
@@ -408,7 +408,9 @@ def _fetch_curl(self, url):
                 partial_file,
             ]  # use a .part file
         else:
-            save_args = ["-O"]
+            # -J flag requires curl >= 7.21.2 ensures file will be named
+            # from redirect header if relevant
+            save_args = ["-O", "-J"]
```

Comment:
* Is `rename_downloaded_executable` still needed in the `wdk` package with this?
* Currently `bootstrap.py` doesn't appear to check `curl` version, but changes which bump the minimum required version can be problematic. This might be reasonable but IMO we could still use a check to provide a clear error message when `curl` is insufficient (and in case users report incompatibilities, we may want to only conditionally do this on Windows in a follow-up PR)